### PR TITLE
assignment_1 is done

### PR DESCRIPTION
**Pull request by RezaDSI**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Refactoring existing code to improve readability and efficiency.
## What did you learn from the changes you have made?
Enhanced understanding of KNN hyperparameter tuning.
## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Considering the use of different classification algorithms like SVM.
## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Yes, encountered data splitting errors which were resolved by correctly defining variables.
## How were these changes tested?
By evaluating the model’s accuracy on the test set.
## A reference to a related issue in your repository (if applicable)
No related issue in the repository.
## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by PatelVishakh — APPROVED** [review]
Assignment 1: Complete. Great work!

Few comments:

For Q1)II) and Q1)III) you should use code to get the desired output.

Note that your train is suppose to be 0.75*178=134 but rather is 140 not entirely sure why this is as the code is fine and output is reasonable. so requires further looking into on my part.

**Comment by RezaDSI** [comment]
Hi Dear PatelVishakh
Thanks for your feedback.
By the way, I'll review Status and reason.
Cheers
Reza
